app/src/main/java/com/example/parkingsystem/mvp/view/ParkingReservationView.kt

```diff
@@ -0,0 +1,66 @@
+package com.example.parkingsystem.mvp.view
+
+import android.app.Activity
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
+import android.icu.util.Calendar
+import com.example.parkingsystem.R
+import com.example.parkingsystem.databinding.ActivityParkingReservationBinding
+import com.example.parkingsystem.mvp.contract.ParkingReservationContract
+import com.example.parkingsystem.mvp.view.base.ActivityView
+import com.example.parkingsystem.util.showToast
+import java.util.Calendar.getInstance
+
+class ParkingReservationView(activity: Activity, private val binding: ActivityParkingReservationBinding) :
+    ParkingReservationContract.ParkingReservationView, ActivityView(activity) {
+
+    override fun getButtonPickerBegin(): Boolean = binding.editDateBegin.isPressed
+
+
+    override fun showDatePickerDialog(dateListener: DatePickerDialog.OnDateSetListener) {
+        val calendar = getInstance()
+        val day = calendar.get(Calendar.DAY_OF_MONTH)
+        val month = calendar.get(Calendar.MONTH)
+        val year = calendar.get(Calendar.YEAR)
+        context?.let {
+            val datePicker = DatePickerDialog(it, dateListener, year, month, day)
+            datePicker.show()
+        }
+    }
+
+    override fun showTimePickerDialog(timeListener: TimePickerDialog.OnTimeSetListener) {
+        val calendar = getInstance()
+        val hour = calendar.get(Calendar.HOUR_OF_DAY)
+        val minute = calendar.get(Calendar.MINUTE)
+        context?.let {
+            val timePicker = TimePickerDialog(it, timeListener, hour, minute, false)
```

Comment:
I think that the TimePickerDialog do not need these initial values, by default load the actual time

Reply:
As we talk, I've made the changes